strawberryfields/apps/qchem/utils.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This module contains various utility functions needed to perform quantum chemistry calculations with
+Strawberry Fields.
+"""
+from typing import Tuple
+
+import numpy as np
+from scipy.constants import c, h, m_u, pi
+
+
+def duschinsky(
+    Li: np.ndarray, Lf: np.ndarray, ri: np.ndarray, rf: np.ndarray, wf: np.ndarray, m: np.ndarray
+) -> Tuple[np.ndarray, np.ndarray]:
+    r"""Generate the Duschinsky rotation matrix and displacement vector.
+
+    The Duschinsky transformation relates the normal coordinates of the initial and
+    final states in a vibronic transition, :math:`q_i` and :math:`q_f` respectively, as:
+
+    .. math:: q_f = U_d q_i + d.
+
+    where :math:`U_d` is a matrix known as the Duschinsky rotation matrix and the displacement parameter
+    :math:`d` is related to the structural changes of the molecule during a vibronic transition.
+    The normal coordinates of a molecule can be represented in terms of atomic displacements as:
+
+    .. math:: q = L^T \sqrt{m} (r -r_e),
```

Comment:
Python conventions are typically more important than scientific conventions. Variable and input names should be lowercase in python, so I think it's better to use `r`. 

This raises the question of using `li`, `lf` instead of `Li`, `Lf`, but since these are matrices, I think it's clearer to use uppercase.